### PR TITLE
fixes #9 / expire old backups

### LINKED ISSUE #9: Expire old backups
AFAICT this just adds new backups all the time. We should consider deleting old backups after a configurable expiry period.

### PR DESCRIPTION
**Pull request by eyenx**
- Expiration of old backups through the variable "EXPIRE"

### CODE REVIEW

File: src/back8sup.sh [hunk 1]
```diff
@@ -131,3 +132,6 @@ do
   log "INFO done exporting all $KIND in namespace $NS"
 done
 log "INFO done exporting namespace $NS"
+
+log "INFO remove backups older than $EXPIRE days"
+find $DST_FOLDER -mtime +$EXPIRE -type f -delete &>/dev/null
```
Comment (hairmare) [inline src/back8sup.sh:137]: Did you consider a "keep the last x generations" logic rather than a "remove everything older than x" logic?

Comment (eyenx) [reply src/back8sup.sh:137]: no. Why is the other logic better?

Comment (hairmare) [reply src/back8sup.sh:137]: In the "remove everything older than x" case old backups would be removed if the job ever fails for longer than $EXPIRE days. Thus I usually prefer to keep old backup generations and not backups older than x. In the past I've done things like the following:

```bash 
GENERATIONS=30

ls -t -w 1 "$DST_FOLDER/*" | tail -n +$GENERATIONS | xargs rm -f
```

Maybe I'm overthinking this and it's fine as is.

Comment (eyenx) [reply src/back8sup.sh:137]: This is good, but IMHO we should rather check that we do not run the cleanup command  if the job fails, right?

Comment (hairmare) [reply src/back8sup.sh:137]: > This is good, but IMHO we should rather check that we do not run the cleanup command if the job fails, right?

sounds fair

**Comment by eyenx** [comment]
I also had to adjust some stuff for #7

**Comment by eyenx** [comment]
@hairmare your turn.

**Comment by eyenx** [comment]
```
2020-07-22T19:08:16 INFO creating directory /mnt/back8sup/202007221908 for export
2020-07-22T19:08:16 INFO checking token and connection to cluster
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
{
  "major": "1",
  "minor": "14",
  "gitVersion": "v1.14.1-k3s.4",
  "gitCommit": "52f3b42401c93c36467f1fd6d294a3aba26c7def",
  "gitTreeState": "clean",
100   272  100   272    0     0   5440      0 --:--:-- --:--:-- --:--:--  5440
  "buildDate": "2019-04-15T22:13+00:00Z",
  "goVersion": "go1.12.1",
  "compiler": "gc",
  "platform": "linux/amd64"
}2020-07-22T19:08:16 INFO parsing /etc/config.yaml
2020-07-22T19:08:16 INFO /etc/config.yaml parsed
2020-07-22T19:08:16 INFO starting with global export
2020-07-22T19:08:16 INFO starting export for all apiservices
apiservices                                    apiregistration.k8s.io         false        APIService
2020-07-22T19:08:17 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1. into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:17 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1.apps into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:18 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1.authentication.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:18 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1.authorization.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:18 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1.autoscaling into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:18 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1.batch into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:19 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1.coordination.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:19 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1.k3s.cattle.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:19 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1.networking.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:19 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1.rbac.authorization.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:20 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1.scheduling.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:20 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1.storage.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:20 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1beta1.admissionregistration.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:20 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1beta1.apiextensions.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:21 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1beta1.apps into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:21 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1beta1.authorization.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:21 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1beta1.batch into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:22 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1beta1.certificates.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:22 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1beta1.coordination.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:22 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1beta1.extensions into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:23 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1beta1.networking.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:23 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1beta1.node.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:23 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1beta1.policy into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:23 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1beta1.rbac.authorization.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:24 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1beta1.scheduling.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:24 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1beta1.storage.k8s.io into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:24 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v1beta2.apps into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:25 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v2beta1.autoscaling into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:25 INFO exporting non-namespaced apiservice.apiregistration.k8s.io/v2beta2.autoscaling into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:25 INFO starting export for all pvc
2020-07-22T19:08:27 INFO done exporting all pvc
2020-07-22T19:08:27 INFO starting export for all ns
namespaces                        ns                                          false        Namespace
mutatingwebhookconfigurations                  admissionregistration.k8s.io   false        MutatingWebhookConfiguration
validatingwebhookconfigurations                admissionregistration.k8s.io   false        ValidatingWebhookConfiguration
customresourcedefinitions         crd,crds     apiextensions.k8s.io           false        CustomResourceDefinition
podsecuritypolicies               psp          extensions                     false        PodSecurityPolicy
2020-07-22T19:08:28 INFO exporting non-namespaced namespace/default into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:28 INFO exporting non-namespaced namespace/kube-node-lease into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:28 INFO exporting non-namespaced namespace/kube-public into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:28 INFO exporting non-namespaced namespace/kube-system into /mnt/back8sup/202007221908/not-namespaced
2020-07-22T19:08:29 INFO starting export for all po
2020-07-22T19:08:29 INFO exporting pod/back8sup-manual-run1-rmq45 from namespace default into /mnt/back8sup/202007221908/default/po
2020-07-22T19:08:30 INFO exporting pod/back8sup-manual-run1-sbk4g from namespace default into /mnt/back8sup/202007221908/default/po
2020-07-22T19:08:30 INFO exporting pod/back8sup-manual-run1-vh6p9 from namespace default into /mnt/back8sup/202007221908/default/po
2020-07-22T19:08:30 INFO exporting pod/back8sup-manual-run10-zcls6 from namespace default into /mnt/back8sup/202007221908/default/po
2020-07-22T19:08:30 INFO exporting pod/back8sup-manual-run3-ts95r from namespace default into /mnt/back8sup/202007221908/default/po
2020-07-22T19:08:31 INFO exporting pod/back8sup-manual-run30-wrp5f from namespace default into /mnt/back8sup/202007221908/default/po
2020-07-22T19:08:32 INFO exporting pod/coredns-695688789-5km67 from namespace kube-system into /mnt/back8sup/202007221908/kube-system/po
2020-07-22T19:08:32 INFO exporting pod/helm-install-traefik-dqxfw from namespace kube-system into /mnt/back8sup/202007221908/kube-system/po
2020-07-22T19:08:33 INFO exporting pod/svclb-traefik-4f86j from namespace kube-system into /mnt/back8sup/202007221908/kube-system/po
2020-07-22T19:08:33 INFO exporting pod/svclb-traefik-g8fnr from namespace kube-system into /mnt/back8sup/202007221908/kube-system/po
2020-07-22T19:08:33 INFO exporting pod/svclb-traefik-n94jl from namespace kube-system into /mnt/back8sup/202007221908/kube-system/po
2020-07-22T19:08:33 INFO exporting pod/svclb-traefik-qs94x from namespace kube-system into /mnt/back8sup/202007221908/kube-system/po
2020-07-22T19:08:34 INFO exporting pod/traefik-55bd9646fc-64zff from namespace kube-system into /mnt/back8sup/202007221908/kube-system/po
2020-07-22T19:08:34 INFO done exporting all po
2020-07-22T19:08:34 INFO done with global export
2020-07-22T19:08:34 INFO starting export in namespace default
2020-07-22T19:08:34 INFO starting export for all cronjob in namespace default
2020-07-22T19:08:34 INFO exporting cronjob.batch/chart-1595443038-back8sup from namespace default into /mnt/back8sup/202007221908/default/cronjob
2020-07-22T19:08:35 INFO starting export for all job in namespace default
2020-07-22T19:08:35 INFO exporting job.batch/back8sup-manual-run1 from namespace default into /mnt/back8sup/202007221908/default/job
2020-07-22T19:08:35 INFO exporting job.batch/back8sup-manual-run10 from namespace default into /mnt/back8sup/202007221908/default/job
2020-07-22T19:08:36 INFO exporting job.batch/back8sup-manual-run3 from namespace default into /mnt/back8sup/202007221908/default/job
2020-07-22T19:08:36 INFO exporting job.batch/back8sup-manual-run30 from namespace default into /mnt/back8sup/202007221908/default/job
2020-07-22T19:08:36 INFO done exporting all job in namespace default
2020-07-22T19:08:36 INFO done exporting namespace default
2020-07-22T19:08:36 INFO remove backups older than 30 days
```

**Comment by eyenx** [comment]
Deletion does not happen if apiserver isn't reachable, if any of the commands fails, but not if a wrong resources gets defined like `namesacpe`

File: src/back8sup.sh [hunk 2]
```diff
@@ -74,7 +79,7 @@ fi
 TOKEN=$(cat "$TOKEN_FILE")
 
 log "INFO checking token and connection to cluster"
-if ! curl -k "$API_ENDPOINT/version/" -H "Authorization: Bearer ${TOKEN}"
+if ! curl --fail -k "$API_ENDPOINT/version/" -H "Authorization: Bearer ${TOKEN}"
```
Comment (tongpu) [inline src/back8sup.sh:82]:
```suggestion
if ! curl --no-progress-meter --fail -k "$API_ENDPOINT/version/" -H "Authorization: Bearer ${TOKEN}"
```

**Pull request merged**